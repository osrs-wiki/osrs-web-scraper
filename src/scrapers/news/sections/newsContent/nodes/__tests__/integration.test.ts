import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import nodeParser from "../parser";

describe("integration test for trailing breaks", () => {
  test("should have proper spacing between headers and content", () => {
    const html = `
<h1>Community Discord Event</h1>
<h3>JMod Wars</h3>
<p>Join us for another round of JMod Castle Wars action!</p>
<p><b>When?</b></p>
<ul>
  <li>Friday, October 24th</li>
  <li>4:00-5:00 PM BST</li>
</ul>
<p><b>Where?</b></p>
<ul>
  <li>World 522</li>
</ul>
<p><b>What?</b></p>
<p>We're flipping your classic Castle Wars experience.</p>
<p>Two J-Mods will join the fray.</p>
`;

    const root = parse(html);
    const builder = new MediaWikiBuilder();

    root.childNodes.forEach((node) => {
      const result = nodeParser(node, { title: "test" });
      if (result) {
        builder.addContents([result].flat());
      }
    });

    expect(builder.build()).toMatchSnapshot();
  });
});
