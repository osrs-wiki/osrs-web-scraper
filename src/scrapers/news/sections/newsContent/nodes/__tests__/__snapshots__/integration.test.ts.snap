// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test for trailing breaks should have proper spacing between headers and content 1`] = `
"==Community Discord Event==
===JMod Wars===
Join us for another round of JMod Castle Wars action!
'''When?'''

* Friday, October 24th

* 4:00-5:00 PM BST

'''Where?'''

* World 522

'''What?'''
We're flipping your classic Castle Wars experience.
Two J-Mods will join the fray.
"
`;
